File: villager/src/main/java/br/com/zup/charlescd/villager/infrastructure/persistence/BuildRepository.java
/*
 * This Source Code Form is subject to the terms of the
 * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
 * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package br.com.zup.charlescd.villager.infrastructure.persistence;

import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;
import java.util.List;

@ApplicationScoped
public class BuildRepository implements PanacheRepository<BuildEntity> {

    public BuildEntity findById(String id) {
        return find("id", id).firstResult();
    }

    public List<BuildEntity> findBuildsToNotify() {
        return find("callback_status = ?1", CallbackStatus.FAILURE.name()).list();
    }

}

File: moove/domain/src/main/kotlin/br/com/zup/charles/domain/repository/BuildRepository.kt
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package br.com.zup.charles.domain.repository

import br.com.zup.charles.domain.*
import java.util.*

interface BuildRepository {

    fun save(build: Build): Build

    fun updateStatus(id: String, status: BuildStatusEnum)

    fun findById(id: String): Optional<Build>

    fun find(id: String, applicationId: String): Optional<Build>

    fun find(tag: String?, status: BuildStatusEnum?, applicationId: String, page: PageRequest): Page<Build>

    fun delete(build: Build)

    fun saveArtifacts(artifacts: List<ArtifactSnapshot>)
}